### DISCUSSION TITLE
OpenWrt routers on boats

### DISCUSSION THREAD

**Discussion by mgrouch** [bareboat-necessities/lysmarine_gen#434] [Ideas]
OpenWrt routers are running Linux OS. 
LTE cellular data modems on the usually have GPS chip. 

I like the idea of using GPS from the router. 
I’ve set up mine to run KPlex and connect via serial port to GPS chip on LTE modem of my router and serve NMEA 0183 traffic from KPlex.  

I think gpsd can be as well used instead of KPlex for that. 

Routers quite often have USB port as well. 

That USB can be used to connect all kind of marine devices. 

NMEA 0183 with FTDI USB sticks. 
Actisense NMEA 2000 usb gateways.
dAISy AIS USB. 

There are some people who compiled RTL_SDR for OpenWrt. 

Effectively OpenWrt router can serve as GPS and NMEA multiplexer

OpenWrt router consumes less power than raspberry pi. 

It’s also a good candidate to put remote alarms logic. Anchor alarm or bilge alarm. Since router has access to internet those alarms can be sent via WhatsApp or other messaging provider.

**Comment by VladimirKalachikhin** [comment]
https://github.com/VladimirKalachikhin/Galadriel-map?tab=readme-ov-file

**Reply by mgrouch** [reply]
It needs continued integration build via GitHub action which builds proper OpenWrt package so it can be installed on any OpenWrt router easier. 

Thanks

**Reply by VladimirKalachikhin** [reply]
https://github.com/VladimirKalachikhin/Galadriel-map/tree/master/emergencykit

I think over-automation is harmful. For some routers, you will need to make special firmware.
https://github.com/VladimirKalachikhin/MT7620_openwrt_firmware
